### PR TITLE
Fixes #22200 - add rest timeout param

### PR DESCRIPTION
**Pull request by chris1984**

### CODE REVIEW

**Comment by stbenjam** [comment]
It'd be nice to get a test in katello_application_spec.rb, ther'es already some coverage of katello.yaml.

**Comment by chris1984** [comment]
@stbenjam added a test, can i get another review please.

File: manifests/application.pp [hunk 1]
```diff
@@ -16,6 +16,8 @@
   Optional[Integer[0, 65535]] $proxy_port = $::katello::proxy_port,
   Optional[String] $proxy_username = $::katello::proxy_username,
   Optional[String] $proxy_password = $::katello::proxy_password,
+  Integer[0] $rest_client_timeout = $::katello::params::rest_client_timeout,
+
```
Comment (ekohl) [inline manifests/application.pp]: Nit: can you remove this line?

File: manifests/init.pp [hunk 2]
```diff
@@ -103,7 +105,7 @@
   String $repo_version = $::katello::params::repo_version,
   Boolean $repo_gpgcheck = $::katello::params::repo_gpgcheck,
   Optional[String] $repo_gpgkey = $::katello::params::repo_gpgkey,
-
+  Integer[0] $rest_client_timeout = $::katello::params::rest_client_timeout,
```
Comment (ekohl) [inline manifests/init.pp]: The settings were grouped. This block is for candlepin. Could you move this to the block with post_sync_token?

File: spec/classes/katello_application_spec.rb [hunk 3]
```diff
@@ -168,6 +196,8 @@
         end
       end
 
+      context
```
Comment (ekohl) [inline spec/classes/katello_application_spec.rb]: This looks like it's not needed.

**Comment by chris1984** [comment]
@ekohl updated with your corrections.

File: spec/classes/katello_application_spec.rb [hunk 4]
```diff
@@ -104,6 +105,33 @@
           end
         end
 
+        context 'with rest client timeout' do
+          let :params do
+            base_params.merge(:rest_client_timeout => 3600)
```
Comment (ekohl) [inline spec/classes/katello_application_spec.rb]: A better test would be to change the value to something else so you know it's not accidentally using a hardcoded value.

**Comment by ekohl** [comment]
Thanks!

**Pull request merged**